File: src/main/java/ru/timakden/chesspuzzle/Queen.java
package ru.timakden.chesspuzzle;

/**
 * Ферзь, перемещается на любое число свободных полей в любом направлении.
 */
class Queen extends ChessPiece {
	Queen(int row, int column) {
		super(row, column);
	}

	@Override
	public String toString() {
		return "Q";
	}

	public boolean canAttackAnotherChessPiece(ChessPiece anotherChessPiece) {
		int row = getRow();
		int column = getColumn();
		int anotherRow = anotherChessPiece.getRow();
		int anotherColumn = anotherChessPiece.getColumn();

		return (Math.abs(row - anotherRow) == Math.abs(column - anotherColumn)) || (row == anotherRow) || (column == anotherColumn);
	}
}

File: src/main/kotlin/ru/timakden/chesspuzzle/Queen.kt
package ru.timakden.chesspuzzle

/**
 * Ферзь, перемещается на любое число свободных полей в любом направлении.
 */
class Queen(row: Int, column: Int) : ChessPiece(row, column) {
    override fun toString() = "Q"

    override fun canAttackAnotherPiece(anotherPiece: ChessPiece): Boolean {
        return Math.abs(row - anotherPiece.row) == Math.abs(column - anotherPiece.column) ||
                row == anotherPiece.row ||
                column == anotherPiece.column
    }
}
